### ISSUE TITLE
Manual grading: allow different order for grading of instance questions

### ISSUE THREAD

**Issue by jonatanschroeder** [PrairieLearn/PrairieLearn#7551]
Currently, when an instance question is submitted, the next submission loaded for the grader is based on a fixed order. Ideally it should be possible for this grading to be done in other orders (e.g., by auto points, or by previous grader, or by submission date).

**Comment by jonatanschroeder** [comment]
As per https://record.umich.edu/articles/study-alphabetical-order-of-surnames-may-affect-grading/, it might be a good idea to default to random.

**Comment by mwest1066** [comment]
For a stable randomish sort we could use something like
```
ORDER BY ((assessment_instance.id % 20369) * 7919) % 10007, assessment_instance.id
```
(with three somewhat-arbitrarily selected primes).

Of course, I'm not sure whether it matters whether we have a stable sort or not.

**Comment by jonatanschroeder** [comment]
I don't think we need stable sort.

**Comment by jonatanschroeder** [comment]
Another option for stable sort, if we decide to use it: `ORDER BY MD5(instance_question.id::TEXT)`. An argument for stable sort in the manual grading assessment question page would be to list the instances in the assessment question in a stable order, in the sense that refreshing the page (or the table) would return the values in the same order, avoiding confusion if staff members change a score and the page refreshes in a weird order. This doesn't need to be followed in the "Grade next instance" case, but could be applied there too.

I think using the instance question ID is better than the assessment instance ID because the latter would be stable across questions, and if we want to avoid biases related to ordering we'd probably want different questions to have different orders so different students get a benefit or disadvantage based on order.

**Comment by mwest1066** [comment]
I agree that using a stable sort would be generally more user-friendly, especially on the overall list page as you say. I agree that using `instance_question.id` is the right thing to do here. I agree that MD5 would also do what we want, but it's more expensive and I don't see any advantage over a simple modulus scheme.
